lib/src/mocks/http.dart

```diff
@@ -68,27 +77,59 @@ handleMockRequest(MockBaseRequest request) {
     return;
   }
 
-  if (_handlers.containsKey(_getUriKey(request.uri))) {
-    var reqHandlers = _handlers[_getUriKey(request.uri)];
+  var matchingRequestHandlerKey = _requestHandlers.keys.firstWhere((key) {
+    return key == _getUriKey(request.uri);
+  }, orElse: () => null);
+
+  Match match;
+  var matchingPatternRequestHandlerKey =
+      _patternRequestHandlers.keys.firstWhere((pattern) {
+    var matches = pattern.allMatches(request.uri.toString());
+    if (matches.isNotEmpty) {
+      match = matches.first;
+      return true;
+    }
+    return false;
+  }, orElse: () => null);
+
+  var handlersByMethod = [];
+  if (matchingRequestHandlerKey != null) {
+    handlersByMethod = _requestHandlers[matchingRequestHandlerKey];
+  } else if (matchingPatternRequestHandlerKey != null) {
+    handlersByMethod =
+        _patternRequestHandlers[matchingPatternRequestHandlerKey];
+  }
 
+  if (handlersByMethod.isNotEmpty) {
     /// Try to find an applicable handler.
     var handler;
-    if (reqHandlers.containsKey(request.method)) {
-      handler = reqHandlers[request.method];
-    } else if (reqHandlers.containsKey('*')) {
-      handler = reqHandlers['*'];
+    if (handlersByMethod.containsKey(request.method)) {
+      handler = handlersByMethod[request.method];
+    } else if (handlersByMethod.containsKey('*')) {
+      handler = handlersByMethod['*'];
     }
 
     /// If a handler was set up for this type of request, call the handler.
     if (handler != null) {
-      request.onSent.then((FinalizedRequest finalizedRequest) {
-        handler(finalizedRequest).then((response) {
-          request.complete(response: response);
-        }, onError: (error) {
-          request.completeError(error: error);
+      if (handler is RequestHandler) {
+        request.onSent.then((FinalizedRequest finalizedRequest) {
+          handler(finalizedRequest).then((response) {
+            request.complete(response: response);
+          }, onError: (error) {
+            request.completeError(error: error);
+          });
         });
-      });
-      return;
+        return;
+      } else if (handler is PatternRequestHandler) {
```

Comment:
Yeah, it could just be an else, I used an else if to be more explicit, but it doesn't really matter either way.